docs: Updating Responsive Styles Support Section of V8 Guide (#1853)
Workday/canvas-kit: modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx

### Responsive Styles

The following Canvas Kit responsive utilities will be released as part of v8:

- **ResponsiveContextProvider**: This is a React Context Provider that will take a width and an

>   optional theme and allow components to subscribe to context changes via the `useResponsiveContext`

Reviewer: ```suggestion
  optional theme prop and allow components to subscribe to context changes via the `useResponsiveContext`
```

Author: Updated

  hook.

- **useResponsiveContext Hook**: This hook subscribes to the `ResponsiveContextProvider` and
  provides helpful utilities to determine whether the current width is within a particular theme

>   breakpoint. Developers can use these utilities when creating container-based responsive styles.

Reviewer: should we reference what those  default theme breakpoints are?

Author: Yes we should. I'll update

### Responsive Styles

> The following Canvas Kit responsive utilities will be released as part of v8:

Author: This makes a lot of sense. I like the link to the Responsive.stories

- **ResponsiveContextProvider**: This is a React Context Provider that will take a width and an

>   optional theme prop and allow components to subscribe to context changes via the

Author: Yeah I was wondering if I should use backticks in that particular part, thanks for clarifying!

  `useResponsiveContext` hook.

> - **useResponsiveContext Hook**: This hook subscribes to the `ResponsiveContextProvider` and
>   provides helpful utilities to determine whether the current width is within a particular theme
>   breakpoint (`Small: 320px, Medium: 768px, Large: 1024px, Extra Large: 1440px`). Developers can use
>   these utilities when creating container-based responsive styles.

Reviewer 2: Removing an unnecessary "Hook". I agree with Manny that providing more context on breakpoints is a good idea; even better than listing them out would be to link to the breakpoints table in the Responsive Styling doc which is more explicit about when each size is applied based on the start and end of the px ranges.

I find a single value to be insufficient to understand when a particular size is applied whereas a range is more clear (i.e., `Small (s) is applied from 320px - 767px` is better than `small: 320px`).

```suggestion
- **useResponsiveContext**: This hook subscribes to the `ResponsiveContextProvider` and provides
  helpful utilities to determine whether the current width is within a particular
  [theme breakpoint](/getting-started/for-developers/resources/responsive-styling/#canvas-kit-breakpoint-sizes).
  Developers can use these utilities when creating container-based responsive styles.
```

Note that although this link _won't_ work in Storybook due to the `#`, it _will_ work on the Canvas Site (more context [here](https://github.com/Workday/canvas-kit/pull/1827#discussion_r988387128)).

Alternatively, we could reproduce the whole passage on Canvas Kit Breakpoint Sizes from your Responsive Styling here in the Upgrade Guide...

![Screen Shot 2022-10-19 at 3 15 03 PM](https://user-images.githubusercontent.com/146020/196814874-e877c724-0189-4311-b86d-bbce867db419.png)

... but I'm not sure that's necessary.

> - **useResponsiveContainerStyles**: A hook that allows developers to create container-based
>   responsive styles using style objects. These are net-new utilities that developers can opt-in to
>   start using, but no action is required to uptake this change.

Reviewer 2: The net-new bit doesn't need to be here in the Upgrade Guide.
```suggestion
- **useResponsiveContainerStyles**: A hook that allows developers to create container-based
  responsive styles using style objects.
```

Author: Updated

```tsx
import {Flex, Box} from '@workday/canvas-kit-react/layout';
import {
  useResizeObserver,
  ResponsiveContextProvider,
  useResponsiveContainerStyles,
} from '@workday/canvas-kit-react/common';
const ref = React.useRef(null);
const [width, setWidth] = React.useState(0);
useResizeObserver({
  ref: ref,
  onResize: data => {
    setWidth(data.width || 0);
  },
});
const containerResponsiveStyles = useResponsiveContainerStyles(
  {
    flex: {
      flexDirection: 'column',
      padding: 'm',
      depth: 1,
      borderRadius: 'l',
      zero: {
        backgroundColor: 'Red',
      },
      s: {
        backgroundColor: 'Orange',
      },
      m: {
        backgroundColor: 'Yellow',
      },
      l: {
        backgroundColor: 'Green',
      },
      xl: {
        backgroundColor: 'Blue',
      },
    },
    box: {
      padding: 's',
    },
  },
  width
);
return (
  <ResponsiveContextProvider width={width}>
    <Box ref={ref}>
      <Flex {...containerResponsiveStyles.flex}>
        <Box {...containerResponsiveStyles.box}>Hello World</Box>
      </Flex>
    </Box>
  </ResponsiveContextProvider>
);
```

> 
> This update is specific to Canvas Kit and will not impact existing designs.

Reviewer 2: This also doesn't need to be in the Upgrade Guide.
```suggestion
```

Author: Updated and thanks for letting me know on this!
